src/resources.ts

```diff
@@ -29,31 +30,32 @@ export function useAuthenticatedHttpResource<ResourceT>(
   setResource: (resource: ResourceT) => void;
   reloadResource: (cancelToken?: CancelToken) => void;
 } {
-  const isMountedRef = useRef(true);
   const [resource, setResource] = useState<ResourceT>(initialState);
   const [loading, setLoading] = useState(false);
   const [error, setError] = useState(null as NullableError);
   const { token, signOut } = useAuthentication();
+  const { translate } = useTranslations();
 
   const loadResource = useCallback(
     async (cancelToken?: CancelToken) => {
       if (token) {
-        isMountedRef.current && setError(null);
-        isMountedRef.current && setLoading(true);
+        setError(null);
+        setLoading(true);
         try {
           const response = await fetcher({ token, cancelToken });
-          isMountedRef.current && setResource(response);
+          setResource(response);
+          setLoading(false);
         } catch (error) {
-          if (isAuthenticationError(error)) {
-            return signOut();
+          if (!http.isCancel(error)) {
```

Comment:
Are you sure this doesn't throw the memory leak error any more, even in the tests?
```
Warning: Can't call setState (or forceUpdate) on an unmounted component.
```

Reply:
I see the error exactly once in the tests. Will dig deeper.

Reply:
Seems the error comes from qr-resizing-stuff in `SharingCode` so should not be related.

Reply:
It was because of the heavy mocking we do in identity page tests. We should move this over to nock instead, but didn't do as part of this pr. Added a `cancelToken.throwIfRequested()` to the problematic mock and it removed the issue.